source/client/factories_impl.cc

```diff
@@ -169,15 +169,15 @@ RequestSourceFactoryImpl::create(const Envoy::Upstream::ClusterManagerPtr& clust
     setRequestHeader(*header, option_header.header().key(), option_header.header().value());
   }
 
-  if (options_.requestSource() == "") {
-    return std::make_unique<StaticRequestSourceImpl>(std::move(header));
-  } else {
+  if (options_.requestSource() != "") {
```

Comment:
I'd also wonder why we edited this code.  It seems that we took a statement that was:
```
If something is true{}
else{}
```
and made it
```
if something is not true{}
else{}
```
In general, I think not true, else statements tend to be a tad more confusing.

Reply:
I wanted it this way because the default option is the static one, so since in a later PR I am adding an additional condition, I thought it made sense to make the non functional change here to set that up.

Reply:
In the future, I think it's easier cognitively on the reviewers if you include the change in the PR where it is relevant, rather than in an unrelated PR.

If it's substantial, you could split it into its own PR, with the intent of making the future PR easier.  I don't think that would be warranted on this particular change